File: leetcode/python/lc719-find-k-th-smallest-pair-distance.py
# coding=utf-8
import unittest

"""719. Find K-th Smallest Pair Distance
https://leetcode.com/problems/find-k-th-smallest-pair-distance/description/

<p>Given an integer array, return the k-th smallest <b>distance</b> among all the pairs. The distance of a pair (A, B) is defined as the absolute difference between A and B. </p>

<p><b>Example 1:</b><br />
<pre>
<b>Input:</b>
nums = [1,3,1]
k = 1
<b>Output: 0</b> 
<b>Explanation:</b>
Here are all the pairs:
(1,3) -> 2
(1,1) -> 0
(3,1) -> 2
Then the 1st smallest distance pair is (1,1), and its distance is 0.
</pre>
</p>

<p><b>Note:</b><br>
<ol>
<li><code>2 <= len(nums) <= 10000</code>.</li>
<li><code>0 <= nums[i] < 1000000</code>.</li>
<li><code>1 <= k <= len(nums) * (len(nums) - 1) / 2</code>.</li>
</ol>
</p>
Similar Questions:
  Find K Pairs with Smallest Sums (find-k-pairs-with-smallest-sums)
  Kth Smallest Element in a Sorted Matrix (kth-smallest-element-in-a-sorted-matrix)
  Find K Closest Elements (find-k-closest-elements)
  Kth Smallest Number in Multiplication Table (kth-smallest-number-in-multiplication-table)
  K-th Smallest Prime Fraction (k-th-smallest-prime-fraction)
"""


class Solution(object):
    def smallestDistancePair(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        

    def test(self):
        pass


if __name__ == "__main__":
    unittest.main()

File: leetcode/golang/lc719-find-k-th-smallest-pair-distance.go
package main

import (
	"fmt"
)

/*719. Find K-th Smallest Pair Distance
https://leetcode.com/problems/find-k-th-smallest-pair-distance/description/

Given an integer array, return the k-th smallest **distance** among all the
pairs. The distance of a pair (A, B) is defined as the absolute difference
between A and B.

**Example 1:**  

    
    
    **Input:**
    nums = [1,3,1]
    k = 1
    **Output: 0** 
    **Explanation:**
    Here are all the pairs:
    (1,3) -> 2
    (1,1) -> 0
    (3,1) -> 2
    Then the 1st smallest distance pair is (1,1), and its distance is 0.
    

**Note:**  

  1. `2 <= len(nums) <= 10000`.
  2. `0 <= nums[i] < 1000000`.
  3. `1 <= k <= len(nums) * (len(nums) - 1) / 2`.


Similar Questions:
  Find K Pairs with Smallest Sums (find-k-pairs-with-smallest-sums)
  Kth Smallest Element in a Sorted Matrix (kth-smallest-element-in-a-sorted-matrix)
  Find K Closest Elements (find-k-closest-elements)
  Kth Smallest Number in Multiplication Table (kth-smallest-number-in-multiplication-table)
  K-th Smallest Prime Fraction (k-th-smallest-prime-fraction)
*/
func smallestDistancePair(nums []int, k int) int {
    
}

func main() {
	fmt.Println()
}
